.github/workflows/test.yml

```diff
@@ -130,31 +129,21 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Cache pip
-      uses: actions/cache@v2
-      with:
-        # This path is specific to Ubuntu
-        path: ~/.cache/pip
-        # Look to see if there is a cache hit for the corresponding requirements file
-        key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt', 'doc/requirements.txt') }}
-        restore-keys: |
-          ${{ runner.os }}-pip-
-          ${{ runner.os }}-
     - name: Install dependencies
       run: |
-        make setup-dev
+        python3 -m pip install nox codecov
     - name: Lint files
       run: |
-        make lint
+        python3 -m nox --non-interactive --session lint
     - name: Check format of files
       if: ${{ ( matrix.python-version != 'pypy3' ) }}
       # Currently fail of this job is OK.
       continue-on-error: true
       run: |
-        make check-format
+        python -m nox --non-interactive --session black
```

Comment:
Yes here should python3 be used.
Why don't use nox directly? Which nox is used if python2.7 and python3 is installed?

Reply:
We no longer install Python 2 in the GH Actions so that might not be a concern any more. But I'm not sure if we're using the `python3 -m` style invocation due to some Windows issues.

Reply:
On my MAC there is also a python installed:
```
python --version
Python 2.7.18 
```
But I see that no pip is available for installing nox.
I'll first fix the python to python3 and can check to use nox directly afterwards.

Reply:
Changed to nox.